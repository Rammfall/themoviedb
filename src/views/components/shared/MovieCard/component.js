import React from 'react'
import PropTypes from 'prop-types'
import { Card } from 'antd'

import { imagePath } from 'Constants'

const MovieCard = ({ title, description, image, actions }) => (
  <Card
    hoverable
    cover={(
      <img
        alt='example'
        src={`${imagePath}${image}`}
      />
    )}
    className='top-margin'
    actions={actions}
  >
    <Card.Meta
      title={title}
      description={description}
    />
  </Card>
)

MovieCard.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string,
  actions: PropTypes.arrayOf(PropTypes.element)
}

MovieCard.defaultProps = {
  image: '',
  actions: null
}

export default MovieCard
