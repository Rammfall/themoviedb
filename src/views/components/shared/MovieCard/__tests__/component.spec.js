import React from 'react'
import { shallow } from 'enzyme'

import MovieCard from '../component'

describe('MovieCard', () => {
  it('matches snapshot', () => {
    const props = {
      title: 'test',
      description: 'test',
      image: 'test'
    }
    const component = shallow(<MovieCard {...props} />)

    expect(component).toMatchSnapshot()
  })
})
