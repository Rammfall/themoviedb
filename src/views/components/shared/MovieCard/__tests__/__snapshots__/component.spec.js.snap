// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard matches snapshot 1`] = `
<Card
  actions={null}
  className="top-margin"
  cover={
    <img
      alt="example"
      src="https://image.tmdb.org/t/p/w500/test"
    />
  }
  hoverable={true}
>
  <Meta
    description="test"
    title="test"
  />
</Card>
`;
